Azure.ResourceManager.Core/Resources/LocationContainer.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.ResourceManager.Core.Adapters;
+using Azure.ResourceManager.Resources;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// Represents an Azure geography region where supported resource providers live.
+    /// </summary>
+    public class LocationContainer : OperationsBase
+    {
+        internal LocationContainer(SubscriptionOperations subscriptionOperations)
+            : base(subscriptionOperations.ClientOptions, subscriptionOperations.Id, subscriptionOperations.Credential, subscriptionOperations.BaseUri)
+        {
+        }
+
+        protected override ResourceType ValidResourceType => SubscriptionOperations.ResourceType;
+
+        /// <summary>
+        /// Gets the subscription client.
+        /// </summary>
+        private SubscriptionsOperations SubscriptionsClient => ResourcesClient.Subscriptions;
+
+        /// <summary>
+        /// Gets the Azure subscriptions.
+        /// </summary>
+        /// <returns> Subscription container. </returns>
+        public SubscriptionContainer GetSubscriptionContainer()
+        {
+            return new SubscriptionContainer(ClientOptions, Credential, BaseUri);
+        }
+
+        /// <summary>
+        /// Lists all geo-locations.
+        /// </summary>
+        /// <returns> A collection of location data that may take multiple service requests to iterate over. </returns>
+        /// <exception cref="InvalidOperationException"> <paramref name="subscriptionId"/> is null. </exception>
+        public Pageable<LocationData> List()
+        {
+            if (string.IsNullOrWhiteSpace(Id.Subscription))
```

Comment:
This cannot be null due to construction requirements, you can safely remove this check